### PR TITLE
[SPARK-13505] [ML] add python api for MaxAbsScaler

### PR DESCRIPTION
**Pull request by zlpmichelle**
## What changes were proposed in this pull request?

After SPARK-13028, we should add Python API for MaxAbsScaler.
## How was this patch tested?

unit test

### CODE REVIEW

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 from pyspark.mllib.linalg import _convert_to_vector
 
 __all__ = ['Binarizer', 'Bucketizer', 'CountVectorizer', 'CountVectorizerModel', 'DCT',
-           'ElementwiseProduct', 'HashingTF', 'IDF', 'IDFModel', 'IndexToString', 'MinMaxScaler',
+           'ElementwiseProduct', 'HashingTF', 'IDF', 'IDFModel', 'IndexToString', 'MaxAbsScaler', 'MaxAbsScalerModel', 'MinMaxScaler',
            'MinMaxScalerModel', 'NGram', 'Normalizer', 'OneHotEncoder', 'PCA', 'PCAModel',
```
Comment (hhbyyh) [inline python/pyspark/ml/feature.py]: please adjust style (max length < 100)

File: python/pyspark/ml/feature.py [hunk 2]
```diff
@@ -543,6 +543,71 @@ class IDFModel(JavaModel):
...
+    statistics, which is also known as min-max normalization or Rescaling. The rescaled value for
+    feature E is calculated as,
+
+    Rescaled(e_i) = (e_i - E_min) / (E_max - E_min) * (max - min) + min
+
+    For the case E_max == E_min, Rescaled(e_i) = 0.5 * (max + min)
+
+    Note that since zero values will probably be transformed to non-zero values, output of the
+    transformer will be DenseVector even for sparse input.
+
```
Comment (hhbyyh) [inline python/pyspark/ml/feature.py]: please refer to the comment in scala MaxAbsScaler

File: python/pyspark/ml/feature.py [hunk 3]
```diff
@@ -543,6 +543,71 @@ class IDFModel(JavaModel):
     .. versionadded:: 1.4.0
     """
 
+@inherit_doc
+class MaxAbsScaler(JavaEstimator, HasInputCol, HasOutputCol):
+    """
+    .. note:: Experimental
+
+    Rescale each feature individually to a common range [min, max] linearly using column summary
```
Comment (MLnick) [inline python/pyspark/ml/feature.py]: It appears you've just copied the docstring and test from `MinMaxScaler`. It is not correct. Please could you follow the doc for the Scala version of `MaxAbsScaler`, and create an appropriate docstring test.

File: python/pyspark/ml/feature.py [hunk 4]
```diff
@@ -28,7 +28,7 @@
 from pyspark.mllib.linalg import _convert_to_vector
 
 __all__ = ['Binarizer', 'Bucketizer', 'CountVectorizer', 'CountVectorizerModel', 'DCT',
-           'ElementwiseProduct', 'HashingTF', 'IDF', 'IDFModel', 'IndexToString', 'MinMaxScaler',
+           'ElementwiseProduct', 'HashingTF', 'IDF', 'IDFModel', 'IndexToString', 'MaxAbsScaler', 'MaxAbsScalerModel', 'MinMaxScaler',
```
Comment (MLnick) [inline python/pyspark/ml/feature.py]: Rather add a new line below the existing ones, otherwise the formatting here is off.

**Comment by hhbyyh** [comment]
Thanks for contribution @zlpmichelle.

File: python/pyspark/ml/feature.py (hunk 4 above)
Comment (hhbyyh) [inline python/pyspark/ml/feature.py]: we'd better keep the alphabetical order here.

**Comment by zlpmichelle** [comment]
Thanks @hhbyyh and @MLnick! Updated.

**Comment by hhbyyh** [comment]
Thanks. I have no other concern.

**Comment by mengxr** [comment]
ok to test

**Comment by zlpmichelle** [comment]
Passed all tests. Thanks @mengxr!

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Pull request closed without merging**